### PR TITLE
add lab3 files

### PR DESCRIPTION
**Pull request by animalvgamer**

### CODE REVIEW

**Comment by animalvgamer** [comment]
I left some references to items in the book in the assignment document. Let me know if those need adjustments.

File: lab3/lab3.rst [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+your structures on the MIPS architecture definitions, unless otherwise stated
+in this assignment.
+
+** **DO NOT CHANGE THE GIVEN SUBCIRCUITS IN ANY WAY. DOING SO WILL LEAD TO AN
+AUTOMATIC ZERO.** **
+
+Details
+=======
+
+Your CPU must implement all instructions from Lab 3:
```
Comment (jlperona) [inline lab3/lab3.rst]: You're missing SRL and SLL as new instructions to implement here.

File: lab3/lab3.rst [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+-  Data instructions:
+   ``ADD, ADDI, AND, ANDI, OR, ORI, SLL, SLT, SRL, SUB, XOR``
+
+-  Memory access instructions: ``LW, SW``
+
+-  Control flow instructions: ``BEQ, J, JAL, JR``
+
+The control signals for the ALU are identical to Lab 2, and are reposted
```
Comment (jlperona) [inline lab3/lab3.rst]: Not exactly identical, with SRL and SLL.

File: lab3/lab3.rst (hunk 1 above)
Comment (jlperona) [reply lab3/lab3.rst]: It's also Lab 2 here, not lab 3.

**Review by jlperona — CHANGES_REQUESTED** [review]

**Review by jlperona — CHANGES_REQUESTED** [review]
Still need to list SRL and SLL as instructions to implement. Better to do this explicitly, i.e. "In addition, you need to implement the following new instructions..."

**Review by jlperona — APPROVED** [review]

**Comment by powerjg** [comment]
Thanks @animalvgamer 

Do you still have the hard copies of the labs that I gave you at the beginning of the quarter? I had some notes on there about minor changes. However, it's not needed to post. I'll go ahead and merge this, but if you have time today or this weekend to make the changes I highlighted, that would be great.

**Comment by animalvgamer** [comment]
@powerjg I do have those copies, but I can't find any notes other than the change from 4 to 3, and a mention of git at the end instead of keeping copies of circuits. I will write another line referring to git and make another pull request.

**Comment by powerjg** [comment]
Ok. I thought I had written something about how those tables of hazards
were confusing. We'll fix it next time.

**Pull request merged**